src/r_draw.h

```diff
@@ -23,18 +23,21 @@
 #include "doomtype.h"
 #include "m_fixed.h"
 
-extern lighttable_t *dc_colormap[2];
-extern int      dc_x;
-extern int      dc_yl;
-extern int      dc_yh;
-extern fixed_t  dc_iscale;
-extern fixed_t  dc_texturemid;
-extern int      dc_texheight;    // killough
-extern byte     dc_skycolor;
-
-// first pixel in a column
-extern byte     *dc_source;         
-extern const byte *dc_brightmap;
+typedef struct
+{
+    lighttable_t *colormap[2];
+    int      x;
+    int      yl;
+    int      yh;
+    fixed_t  iscale;
+    fixed_t  texturemid;
+    int      texheight; // killough
+    byte     skycolor;
+    byte     *source; // first pixel in a column
+    const byte *brightmap;
+    byte *translation;
+} dc_t;
+extern dc_t *const dc;
```

Comment:
`dc` is a short name for a global variable. We can make it static in `r_draw.c` and change its values from a function, as in PrBoom+. Alternatively, we can refactor it as local function parameter, which would be helpful for a multi-threaded renderer, like `cb_column_t` in Eternity: https://github.com/team-eternity/eternity/blob/b05622899d44be44b3a28c8fa965c56a084d7a73/source/r_draw.cpp#L179